bridge-fungible-token/src/main.sw

```diff
@@ -1,69 +1,186 @@
 contract;
 
+dep data;
+dep errors;
+dep events;
 dep utils;
 
-use contract_message_receiver::MessageReceiver;
 use bridge_fungible_token_abi::BridgeFungibleToken;
-use std::address::Address;
-use std::contract_id::ContractId;
-use std::constants::ZERO_B256;
-use utils::{input_message_data, input_message_data_length};
+use contract_message_receiver::MessageReceiver;
+use core::num::*;
+// use data::MessageData;
+use errors::BridgeFungibleTokenError;
+use events::WithdrawalEvent;
+use std::{
+    address::Address,
+    assert::assert,
+    chain::auth::{
+        AuthError,
+        msg_sender,
+    },
+    context::{
+        call_frames::{
+            contract_id,
+            msg_asset_id,
+        },
+        msg_amount,
+    },
+    contract_id::ContractId,
+    identity::Identity,
+    logging::log,
+    option::Option,
+    result::Result,
+    revert::{
+        require,
+        revert,
+    },
+    storage::StorageMap,
+    u256::U256,
+    vm::evm::evm_address::EvmAddress,
+};
+use utils::{
+    burn_tokens,
+    correct_input_type,
+    decompose,
+    input_message_data,
+    input_message_data_length,
+    input_message_recipient,
+    input_message_sender,
+    is_address,
+    mint_tokens,
+    parse_message_data,
+    send_message,
+    transfer_tokens,
+};
 
+////////////////////////////////////////
+// Constants
+////////////////////////////////////////
+const NAME = "PLACEHOLDER";
```

Comment:
I always envisioned these as being set in a constructor/initializer somehow as params. It would avoid having to compile this for each token which might make deployment easier

Reply:
Agreed, I've moved them into the manifest already but haven't pushed my code yet.